compiler/rustc_borrowck/src/type_check/free_region_relations.rs

```diff
@@ -320,6 +320,9 @@ impl<'tcx> UniversalRegionRelationsBuilder<'_, 'tcx> {
             .fully_perform(self.infcx, DUMMY_SP)
             .unwrap_or_else(|_| bug!("failed to compute implied bounds {:?}", ty));
         debug!(?bounds, ?constraints);
+        // Because of #109628, we may have unexpected placeholders. Ignore them!
+        // FIXME(#109628): panic in this case once the issue is fixed.
+        let bounds = bounds.into_iter().filter(|bound| !bound.has_placeholders());
```

Comment:
can you move that into the `TypeOp`?

Reply:
Moving this into the TypeOp has no benefit as we would have to duplicate the logic in `QueryTypeOp::{perform_query,perform_locally_in_new_solver}` as well. It is also more difficult to handle canonicalized query responses - there is no easy way to detect placeholders for example.

Doing it in the query itself is difficult as we need to do  `OpportunisticRegionResolution` before checking for placeholders. Using `TypeVisitableExt::has_placeholders` to check for erroneus placeholders there does not play well with #109388 as it would yield many false positives.